src/webgpu/api/operation/rendering/blending.spec.ts

```diff
@@ -279,7 +281,96 @@ g.test('formats')
     `Test blending results works for all formats that support it, and that blending is not applied
   for formats that do not. Blending should be done in linear space for srgb formats.`
   )
-  .unimplemented();
+  .params(u =>
+    u //
+      .combine('format', [
+        'r16float',
+        'bgra8unorm-srgb',
+        'r8unorm',
+        'rg16float',
+        'rgb10a2unorm',
+        'rgba16float',
+        'rgba8unorm',
+      ] as const)
+  )
+  .fn(async t => {
+    const { format } = t.params;
+
+    const pipeline = t.device.createRenderPipeline({
+      layout: 'auto',
+      fragment: {
+        targets: [
+          {
+            format,
+            blend: {
+              color: { srcFactor: 'one', dstFactor: 'one', operation: 'add' },
+              alpha: {},
+            },
+          },
+        ],
+        module: t.device.createShaderModule({
+          code: `
+struct Uniform {
+  color: vec4<f32>
+};
+@group(0) @binding(0) var<uniform> u : Uniform;
+
+@fragment fn main() -> @location(0) vec4<f32> {
+  return u.color;
+}
+          `,
+        }),
+        entryPoint: 'main',
+      },
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+@vertex fn main() -> @builtin(position) vec4<f32> {
+    return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+}
+          `,
+        }),
+        entryPoint: 'main',
+      },
+      primitive: {
+        topology: 'point-list',
+      },
+    });
+
+    const renderTarget = t.device.createTexture({
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+      size: [1, 1, 1],
+      format,
+    });
+
+    const commandEncoder = t.device.createCommandEncoder();
+    const renderPass = commandEncoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: renderTarget.createView(),
+          clearValue: { r: 0.2, g: 0.2, b: 0.2, a: 0.2 },
+          loadOp: 'clear',
+          storeOp: 'store',
+        },
+      ],
+    });
+    renderPass.setPipeline(pipeline);
+
+    const expColor = { R: 0.6, G: 0.6, B: 0.6, A: 0.6 };
+    const expTexelView = TexelView.fromTexelsAsColors(format, coords => expColor);
+
+    const result = await textureContentIsOKByT2B(
+      t,
+      { texture: renderTarget },
+      [1, 1, 1],
+      { expTexelView },
+      {
+        maxDiffULPsForNormFormat: 614,
```

Comment:
In `norm' format cases, the diff generates up to 614. And, in 'float' format cases, the diff happened up to 13517. It looks like I'm missing setting something in the test. Trying to find them.

Reply:
FYI, these are the error messages when  `maxDiffULPsForNormFormat` and `  maxDiffULPsForFloatFormat:` are `1`.

```
$ ./src/dawn/node/tools/run-cts --cts=/home/gyuyoung/github/cts --dawn-node=/home/gyuyoung/chromium/dawn/out/cmake-debug/dawn.node --backend=default --output=/home/gyuyoung/webgpu-result.txt 'webgpu:api,operation,rendering,blending:formats:*'
Running in parallel with server...
Testing 7 test cases...
webgpu:api,operation,rendering,blending:formats:format="rgb10a2unorm" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords ==   X,Y,Z:                               0,0,0
      act. texel bytes (little-endian) ==      0x:                         00 00 00 00
                           act. colors == R,G,B,A:     0.00000,0.00000,0.00000,0.00000
                           exp. colors == R,G,B,A: 0.600000,0.600000,0.600000,0.600000
            act. normal-ULPs-from-zero == R,G,B,A:                             0,0,0,0
            exp. normal-ULPs-from-zero == R,G,B,A:                       614,614,614,2
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==                            -614,-614,-614,-2
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


webgpu:api,operation,rendering,blending:formats:format="rgba8unorm" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords ==   X,Y,Z:                               0,0,0
      act. texel bytes (little-endian) ==      0x:                         00 00 00 00
                           act. colors == R,G,B,A:     0.00000,0.00000,0.00000,0.00000
                           exp. colors == R,G,B,A: 0.600000,0.600000,0.600000,0.600000
            act. normal-ULPs-from-zero == R,G,B,A:                             0,0,0,0
            exp. normal-ULPs-from-zero == R,G,B,A:                     153,153,153,153
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==                          -153,-153,-153,-153
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)
webgpu:api,operation,rendering,blending:formats:format="r8unorm" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords == X,Y,Z:    0,0,0
      act. texel bytes (little-endian) ==    0x:       00
                           act. colors ==     R:  0.00000
                           exp. colors ==     R: 0.600000
            act. normal-ULPs-from-zero ==     R:        0
            exp. normal-ULPs-from-zero ==     R:      153
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==            -153
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


webgpu:api,operation,rendering,blending:formats:format="bgra8unorm-srgb" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords ==   X,Y,Z:                               0,0,0
      act. texel bytes (little-endian) ==      0x:                         00 00 00 00
                           act. colors == B,G,R,A:     0.00000,0.00000,0.00000,0.00000
                           exp. colors == B,G,R,A: 0.600000,0.600000,0.600000,0.600000
            act. normal-ULPs-from-zero == B,G,R,A:                             0,0,0,0
            exp. normal-ULPs-from-zero == B,G,R,A:                     203,203,203,153
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==                          -203,-203,-203,-153
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


webgpu:api,operation,rendering,blending:formats:format="rg16float" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords == X,Y,Z:             0,0,0
      act. texel bytes (little-endian) ==    0x:       00 00 00 00
                           act. colors ==   R,G:   0.00000,0.00000
                           exp. colors ==   R,G: 0.600000,0.600000
            act. normal-ULPs-from-zero ==   R,G:               0,0
            exp. normal-ULPs-from-zero ==   R,G:       13517,13517
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==            -13517,-13517
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


webgpu:api,operation,rendering,blending:formats:format="r16float" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords == X,Y,Z:    0,0,0
      act. texel bytes (little-endian) ==    0x:    00 00
                           act. colors ==     R:  0.00000
                           exp. colors ==     R: 0.600000
            act. normal-ULPs-from-zero ==     R:        0
            exp. normal-ULPs-from-zero ==     R:    13517
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==          -13517
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


webgpu:api,operation,rendering,blending:formats:format="rgba16float" - fail:
  - EXPECTATION FAILED: Texture level had unexpected contents:
     between 0,0,0 and 0,0,0 inclusive:
                                coords ==   X,Y,Z:                               0,0,0
      act. texel bytes (little-endian) ==      0x:             00 00 00 00 00 00 00 00
                           act. colors == R,G,B,A:     0.00000,0.00000,0.00000,0.00000
                           exp. colors == R,G,B,A: 0.600000,0.600000,0.600000,0.600000
            act. normal-ULPs-from-zero == R,G,B,A:                             0,0,0,0
            exp. normal-ULPs-from-zero == R,G,B,A:             13517,13517,13517,13517
             tolerance ± 1 normal-ULPs
       diff (act - exp) in normal-ULPs ==                  -13517,-13517,-13517,-13517
        at textureContentIsOKByT2B (/home/gyuyoung/github/cts/out-node/webgpu/util/texture/texture_ok.js:234:12)
        at async RunCaseSpecific.fn (/home/gyuyoung/github/cts/out-node/webgpu/api/operation/rendering/blending.spec.js:298:20)Warning: Backend validation enabled but Dawn was not built with DAWN_ENABLE_VULKAN_VALIDATION_LAYERS.
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0


⣷ [▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▏] 100.0%
Completed in 3.796839594s
    FAIL: 7         (100.0%)
```

Reply:
The symptom as shown in the error is that the render target is all 0s instead of having some rendering result in it.

Now that I've had a chance to look more closely, the reason is that the command buffer is not submitted (and it's also missing a draw call).